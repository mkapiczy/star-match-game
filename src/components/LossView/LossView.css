.bomb-wrapper {
    display: flex;
    position: absolute;
    top: 0;
    left: 0;
    width: 100%;
    height: 100%;
    justify-content: center;
    align-content: center;
}

.bomb-image {
    top: 15vmin;
    position: absolute;
    width: 40vmin;
    height: auto;
    z-index: 1;
}

.title-wrapper {
    position: absolute;
    display: flex;
    flex-direction: column;
    z-index: 2;
    justify-content: center;
    align-content: center;
}

.game-over-title {
    color: white;
    font-weight: bold;
    font-size: 4vmin;
}

.bomb-play-again-button {
    width: 10vmin;
    margin: auto;
    font-size: 2vmin;
}


