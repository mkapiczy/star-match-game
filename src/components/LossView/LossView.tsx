import React from 'react';
import './LossView.scss';
import bomb from '../../assets/bomb.gif'

interface Props {
    onClick: any
}

const LossView: React.FC<Props> = (props: Props) => {

    return (
        <div className='bomb-wrapper'>
            <img className={"bomb-image"} src={bomb} alt="Bomb gif"/>
            <div className="title-wrapper">
                <div className="game-over-title">Game Over!</div>
                <button className="bomb-play-again-button" onClick={props.onClick}>Play Again</button>
            </div>
        </div>
    );
}

export default LossView;