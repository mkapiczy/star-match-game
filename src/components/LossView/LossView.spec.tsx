import React from 'react';
import LossView from "./LossView";
import renderer from 'react-test-renderer';

describe("LossView component snapshot tests", () => {
    it("should match snapshot", () => {
        const component = renderer.create(
            <LossView onClick={() => ''}/>
        );
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    })
})