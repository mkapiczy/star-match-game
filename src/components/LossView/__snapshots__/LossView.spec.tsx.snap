// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LossView component snapshot tests should match snapshot 1`] = `
<div
  className="bomb-wrapper"
>
  <img
    alt="Bomb gif"
    className="bomb-image"
    src="bomb.gif"
  />
  <div
    className="title-wrapper"
  >
    <div
      className="game-over-title"
    >
      Game Over!
    </div>
    <button
      className="bomb-play-again-button"
      onClick={[Function]}
    >
      Play Again
    </button>
  </div>
</div>
`;
